### PR TITLE
[Merged by Bors] - feature(http-client): add async-std compat layer and hyper connector

### PR DESCRIPTION
**Pull request by matheus-consoli**
Part 1 of the `mini-http-client` implementation.

This PR updates some dependencies and adds the `async-std` and `hyper` compatibility building blocks.

### CODE REVIEW

File: src/http_client/async_std_compat.rs [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+
+impl CompatConnector {
+    pub fn new(tls_config: ClientConfig) -> Self {
+        Self(Arc::new(TlsConnector::from(Arc::new(tls_config))))
+    }
+}
+
+impl Service<Uri> for CompatConnector {
+    type Response = TlsStream;
+    type Error = ConnectorError;
```
Comment (sehz) [inline src/http_client/async_std_compat.rs:50]: Use anyhow to reduce error wrapping

**Review by sehz — CHANGES_REQUESTED** [review]
Overall looks good.

Couple of issues:
- return anyhow error
- need to bump up crate version
- integration test.   there are few network tests as example

**Comment by matheus-consoli** [comment]
@sehz, I updated the crate version and we're now using `anyhow` for error handling

there are integration tests in the second part of this feature (#216)

**Comment by sehz** [comment]
Since this is feature, it seems only need patch version change?

**Comment by digikata** [comment]
It looks like that change is already there 0.6 -> 0.7?

File: Cargo.toml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "fluvio-future"
-version = "0.6.0"
+version = "0.7.0"
```
Comment (sehz) [inline Cargo.toml:3]: should be 0.6.1

**Review by sehz — CHANGES_REQUESTED** [review]

File: Cargo.toml (hunk 2 above)
Comment (matheus-consoli) [reply Cargo.toml:3]: but semver.org says new feature = minor

Comment (sehz) [reply Cargo.toml:3]: only for release version (> 1.0.0) otherwise bumping to 0.7 is considered breaking change which this is not

Comment (matheus-consoli) [reply Cargo.toml]: ok, changed to 6.1

**Review by sehz — APPROVED** [review]

**Pull request closed without merging**